tests/console/login.go

```diff
@@ -108,41 +108,53 @@ func LoginToFedora(vmi *v1.VirtualMachineInstance) error {
 	}
 	defer expecter.Close()
 
-	b := append([]expect.Batcher{
-		&expect.BSnd{S: "\n"},
-		&expect.BSnd{S: "\n"},
-		&expect.BCas{C: []expect.Caser{
-			&expect.Case{
-				// Using only "login: " would match things like "Last failed login: Tue Jun  9 22:25:30 UTC 2020 on ttyS0"
-				// and in case the VM's did not get hostname form DHCP server try the default hostname
-				R:  regexp.MustCompile(fmt.Sprintf(`(localhost|%s) login: `, vmi.Name)),
-				S:  "fedora\n",
-				T:  expect.Next(),
-				Rt: 10,
-			},
-			&expect.Case{
-				R:  regexp.MustCompile(`Password:`),
-				S:  "fedora\n",
-				T:  expect.Next(),
-				Rt: 10,
-			},
-			&expect.Case{
-				R:  regexp.MustCompile(`Login incorrect`),
-				T:  expect.LogContinue("Failed to log in", expect.NewStatus(codes.PermissionDenied, "login failed")),
-				Rt: 10,
-			},
-			&expect.Case{
-				R: regexp.MustCompile(fmt.Sprintf(`\[fedora@%s ~\]\$ `, vmi.Name)),
-				T: expect.OK(),
-			},
-		}},
-		&expect.BSnd{S: "sudo su\n"},
-		&expect.BExp{R: PromptExpression},
-	})
-	res, err := expecter.ExpectBatch(b, 3*time.Minute)
-	if err != nil {
-		log.DefaultLogger().Object(vmi).Infof("Login: %+v", res)
-		return err
+	var loginErr error
+	// Sometimes the login promped is ripped apart by asynchronous service messages on the first prompt, most of the time
+	// from cloud-init. As a soltuion. instead of waiting three minutes for the VM to boot, only wait one minute and
+	// if we did not login, send newlines and check again.
+	for retries := 0; retries < 3; retries++ {
```

Comment:
You could use `wait.PollImmediate` here.

Reply:
+1

Reply:
I tried now a different solution, where I only retry once after two minutes. I was not quite happy about not getting the whole boot log anymore so nicely together like we do right now. Maybe this is already good enough. What do you think?

Reply:
If it works it feels smaller although we can implement it with "wait" package primitives.

Reply:
Yes, it just feels more on-spot. I did a version with wait too. I don't mind too much which one we take.